### ISSUE TITLE
Incoherent behavior in initialization from dict with tuple keys

### ISSUE THREAD

**Issue by toobaz** [pandas-dev/pandas#18484]
#### Code Sample, a copy-pastable example if possible

```python
In [2]: d = {(0, 1) : 2, (3, 4) : 5}

In [3]: pd.Index(d)
Out[3]: Index([(0, 1), (3, 4)], dtype='object')

In [4]: pd.Index(list(d))
Out[4]: 
MultiIndex(levels=[[0, 3], [1, 4]],
           labels=[[0, 1], [0, 1]])

In [5]: pd.Series(d).index
Out[5]: 
MultiIndex(levels=[[0, 3], [1, 4]],
           labels=[[0, 1], [0, 1]])

In [6]: pd.Series(index=list(d)).index
Out[6]: Index([(0, 1), (3, 4)], dtype='object')

```
#### Problem description

I guess ``Out[3]:`` and ``Out[6]:`` are wrong.

#### Expected Output

``Out[4]:``

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.0-4-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8
LOCALE: en_GB.UTF-8

pandas: 0.22.0.dev0+202.g97bd66ea8
pytest: 3.0.6
pip: 9.0.1
setuptools: 33.1.1
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.18.1
pyarrow: None
xarray: None
IPython: 5.2.2
sphinx: None
patsy: 0.4.1+dev
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: 1.2.0
tables: 3.3.0
numexpr: 2.6.1
feather: 0.3.1
matplotlib: 2.0.0
openpyxl: 2.3.0
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: None
lxml: 3.7.1
bs4: 4.5.3
html5lib: 0.999999999
sqlalchemy: 1.0.15
pymysql: None
psycopg2: None
jinja2: 2.8
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

**Comment by toobaz** [comment]
[This](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/series/test_apply.py#L418) suggests the above is partly indended... but there should be a better fix.

**Comment by jreback** [comment]
```
In [16]: pd.Index(list(d), tupleize_cols=True)
Out[16]: 
MultiIndex(levels=[[0, 3], [1, 4]],
           labels=[[0, 1], [0, 1]])

In [17]: pd.Index(list(d), tupleize_cols=False)
Out[17]: Index([(0, 1), (3, 4)], dtype='object')
```

**Comment by jreback** [comment]
I hate this kw.

**Comment by jreback** [comment]
@TomAugspurger

**Comment by toobaz** [comment]
It gets better:
```python
In [2]: pd.Index([[0, 1, 2], [3, 4, 5]])
Out[2]: Index([[0, 1, 2], [3, 4, 5]], dtype='object')

In [3]: pd.Series(index=[[0, 1, 2], [3, 4, 5]])
Out[3]: 
0  3   NaN
1  4   NaN
2  5   NaN
dtype: float64
```
... which is _not_ "tupleization" (notice the orientation)!

The interpretation of lists as index levels in ``Series`` (and ``DataFrame``) is totally unexpected to me... but since it is there, maybe we want to support it in ``Index`` too?

**Comment by jreback** [comment]
[2] is invalid but prob not checked

these are not hashable sub elements 

absolutely we do not want to add complexity like this

**Comment by jorisvandenbossche** [comment]
Yes, and I didn't know this worked like that in Series. I would rather disallow it there as well.

**Comment by toobaz** [comment]
I'm totally fine with disallowing this feature, but it is precisely the [main task](https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/lib.pyx#L746) of ``lib.clean_index_list``, and it is [tested](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexes/period/test_period.py#L647), so it's not accidental.

(and assuming we didn't want to drop it, integrating it into ``Index`` would actually simplify the code, which is why I was suggesting it... but again, if you want to drop it, good)

**Comment by jorisvandenbossche** [comment]
> and it is tested, so it's not accidental.

It's not accidental, that's true, but that test is more testing that the Period is preserved and not converted into an int, than testing the "index levels as list" feature I think. Although there are maybe other more explicit tests.

**Comment by toobaz** [comment]
By the way: ``In [3]:`` was actually fixed by #18514 (not ``In [6]``). Retitling accordingly.

**Comment by jreback** [comment]
[3] is pretty clear. We coerce this to a MultiIndex. [5] should work as well (to a MI). I see why you think https://github.com/pandas-dev/pandas/issues/18484#issuecomment-347695068 is inconsistent.

are:

a) ``pd.Index([[....], [....]])``

b) ``infer_to_index([[...], [....]])``

the same, where ``infer_to_index`` happens upon assignment (e.g. [3]). I actually could buy that these infer to the same, namely a MultiIndex (an Index with nested elements is completely non-supported however)

**Comment by toobaz** [comment]
OK. So finally if I understand correctly the plan is to
- make ``pd.Index([[0, 1, 2], [3, 4, 5]])`` return a ``MultiIndex`` (without any argument in the vein of ``tupleize_cols=False``, since anyway we don't want lists as labels)
- suppress ``_ensure_index``, which becomes redundant (since it seems to me it isn't catching any other case)

**Comment by jreback** [comment]
> suppress _ensure_index, which becomes redundant (since it seems to me it isn't catching any other case)

the original intent of ``_ensure_index`` was I think to infer whether things should be an Index or a MultiIndex. If you can remove code great.
